src/readsbml.jl

```diff
@@ -96,6 +96,39 @@ function readSBML(fn::String)::SBML.Model
             throw(AssertionError("SBML document contains no model"))
         end
 
+        for (converter, kwargs) in conversion_options
+            if converter == "setLevelAndVersion"
+                success = ccall(sbml(:SBMLDocument_setLevelAndVersion), Cint, (VPtr,Cint,Cint), doc, kwargs["level"], kwargs["version"])
+                continue
+            end
+            props = ccall(sbml(:ConversionProperties_create), VPtr, ())
+            option = ccall(sbml(:ConversionOption_create), VPtr, (Cstring,), converter)
+            ccall(sbml(:ConversionProperties_addOption), Cvoid, (VPtr, VPtr), props, option)
+            if !(kwargs==nothing)
+                for (k, v) in kwargs
+                    option = ccall(sbml(:ConversionOption_create), VPtr, (Cstring,), k)
+                    ccall(sbml(:ConversionOption_setValue), Cvoid, (VPtr, Cstring), option, v)
+                    ccall(sbml(:ConversionProperties_addOption), Cvoid, (VPtr, VPtr), props, option)
+                end
+            success = ccall(sbml(:SBMLDocument_convert), Cint, (VPtr,VPtr), doc, props)
+            end
+        end
+
+        n_errs = ccall(sbml(:SBMLDocument_getNumErrors), Cuint, (VPtr,), doc)
+        for i = 0:n_errs-1
+            err = ccall(sbml(:SBMLDocument_getError), VPtr, (VPtr, Cuint), doc, i)
+            msg = SBML.get_string(err, :XMLError_getMessage)
+            @warn "SBML reported error after conversion: $msg"
+        end
+        if n_errs > 0  # PL: @Mirek: Maybe we should only throw an error if certain severity is exceeded using `SBMLDocument_getNumErrorsWithSeverity`?
```

Comment:
I hoped that this excludes the "info" and "warning" severities.

(SBML error severities are more like logleves, they don't carry any semantics that would tell you whether it's feasible to continue, e.g. if a data loss was likely. So we likely can't solve this one well.)

Reply:
btw given this is going to be externalized to a function, you can very easily leave that decision on the user.

Reply:
I do not know what severities are excluded, but if I use the converters (for instance the `setLevelAndVersionConverter`) the AssertionError is thown because of the following errors, which I believe are not severe enough to stop the import:
```
┌ Warning: SBML reported error after conversion: In situations where a mathematical expression contains literal numbers or parameters whose units have not been declared, it is not possible to verify accurately the consistency of the units in the expression.
│  The units of the <kineticLaw> <math> expression 'c1 * k1 * s1 * s2' cannot be fully checked. Unit consistency reported as either no errors or further unit errors related to this object may not be accurate.
└ @ SBML C:\Users\wolf5212\OneDrive - Nexus365\DPhil Project\Programmes\juliadev\dev\LCSB-BioCore\SBML.jl\src\readsbml.jl:121
┌ Warning: SBML reported error after conversion: In situations where a mathematical expression contains literal numbers or parameters whose units have not been declared, it is not possible to verify accurately the consistency of the units in the expression.
│  The units of the <kineticLaw> <math> expression 'c1 * k1 * s3' cannot be fully checked. Unit consistency reported as either no errors or further unit errors related to this object may not be accurate.
└ @ SBML C:\Users\wolf5212\OneDrive - Nexus365\DPhil Project\Programmes\juliadev\dev\LCSB-BioCore\SBML.jl\src\readsbml.jl:121
┌ Warning: SBML reported error after conversion: In situations where a mathematical expression contains literal numbers or parameters whose units have not been declared, it is not possible to verify accurately the consistency of the units in the expression.
│  The units of the <kineticLaw> <math> expression 'c1 * function_01_1(s4)' cannot be fully checked. Unit consistency reported as either no errors or further unit errors related to this object may not be accurate.      
└ @ SBML C:\Users\wolf5212\OneDrive - Nexus365\DPhil Project\Programmes\juliadev\dev\LCSB-BioCore\SBML.jl\src\readsbml.jl:121
┌ Warning: SBML reported error after conversion: In SBML's simple symbol system, there is no separation of symbols by class of object; consequently, inside the kinetic law mathematical formula, the value of a local parameter having the same identifier as a species or compartment or other global model entity will override the global value. Modelers may wish to take precautions to avoid this happening accidentally.
│  In this instance the local parameter with id 'k1' will shadow the 'Parameter' with an identical id.
└ @ SBML C:\Users\wolf5212\OneDrive - Nexus365\DPhil Project\Programmes\juliadev\dev\LCSB-BioCore\SBML.jl\src\readsbml.jl:121
┌ Warning: SBML reported error after conversion: As a principle of best modeling practice, the units of a <parameter> should be declared rather than be left undefined. Doing so improves the ability of software to check 
the consistency of units and helps make it easier to detect potential errors in models.
│  The <parameter> with the id 'k1' does not have a 'units' attribute.
└ @ SBML C:\Users\wolf5212\OneDrive - Nexus365\DPhil Project\Programmes\juliadev\dev\LCSB-BioCore\SBML.jl\src\readsbml.jl:121
┌ Warning: SBML reported error after conversion: As a principle of best modeling practice, the units of a <parameter> should be declared rather than be left undefined. Doing so improves the ability of software to check 
the consistency of units and helps make it easier to detect potential errors in models.
│  The <localParameter> with the id 'k1' does not have a 'units' attribute.
└ @ SBML C:\Users\wolf5212\OneDrive - Nexus365\DPhil Project\Programmes\juliadev\dev\LCSB-BioCore\SBML.jl\src\readsbml.jl:121
```

Reply:
So can I get rid of throwing the `AssertionError` for the time being?